### PR TITLE
Fixes write_error loop from bail_out function

### PR DESCRIPTION
**Pull request by seemethere**
Fixes stack-overflow found in #307

### CODE REVIEW

File: sanic/server.py [hunk 1]
```diff
@@ -189,9 +190,15 @@ def write_error(self, exception):
                 "Writing error failed, connection closed {}".format(e))
 
     def bail_out(self, message):
-        exception = ServerError(message)
-        self.write_error(exception)
-        log.error(message)
+        if self.transport.is_closing():
+            log.error(
+                "Connection closed before error was sent to user @ {}".format(
+                    self.transport.get_extra_info('peername')))
```
Comment (r0fls) [inline sanic/server.py]: I think you can use the property here`self.request.ip`

Comment (seemethere) [inline sanic/server.py]: Well in the example that we had seen from the issue self.parser was still None so I'd be wary of if self.request was filled out correctly as well. It's just a safe way of doing it.

**Comment by r0fls** [comment]
@seemethere I was going to use another try/except here. Are there any other exceptions that could occur during `self_write` within `bail_out` and lead to a circular exception? If there are, the try/except makes more sense imo.

**Comment by r0fls** [comment]
I would also think a try/except would be faster than checking the `transport.is_closing()` method each time, but I don't have a working test case for this. The issue didn't actually throw any exception for me ... maybe I did something wrong though.

**Comment by seemethere** [comment]
You might need to run it through wrk to replicate the issue.

**Comment by r0fls** [comment]
I saw almost no benefit to using try/except instead, so let's use this for now.

**Comment by sh0uw4ng** [comment]
[241](https://github.com/channelcat/sanic/issues/241)

**Pull request merged**